tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.Tests/FacebookAdapterTests.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(conversationReference, null); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldSucceed()
+        {
+            var callbackInvoked = false;
+
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+            Task BotsLogic(ITurnContext turnContext, CancellationToken cancellationToken)
+            {
+                callbackInvoked = true;
+                return Task.CompletedTask;
+            }
+
+            await facebookAdapter.ContinueConversationAsync(conversationReference, BotsLogic);
+            Assert.True(callbackInvoked);
+        }
+
+        [Fact]
+        public async System.Threading.Tasks.Task DeleteActivityAsyncShouldThrowNotImplementedException()
+        {
+            var adapter = new FacebookAdapter(new FacebookClientWrapper(_testOptions));
+
+            var activity = new Activity();
+            var turnContext = new TurnContext(adapter, activity);
+
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<NotImplementedException>(() => adapter.DeleteActivityAsync(turnContext, conversationReference, default));
+        }
+
+        [Fact]
+        public async void ProcessAsyncShouldThrowExceptionWithInvalidBody()
+        {
+            var clientWrapper = new Mock<FacebookClientWrapper>(_testOptions);
+            clientWrapper.Setup(x => x.VerifySignature(It.IsAny<HttpRequest>(), It.IsAny<string>())).Returns(true);
+
+            var facebookAdapter = new FacebookAdapter(clientWrapper.Object);
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes("wrong-formatted-json"));
+
+            var httpRequest = new Mock<HttpRequest>();
+            httpRequest.SetupGet(req => req.Query[It.IsAny<string>()]).Returns("test");
+            httpRequest.SetupGet(req => req.Body).Returns(stream);
+            var httpResponse = new Mock<HttpResponse>();
+            var bot = new Mock<IBot>();
+            bot.Setup(x => x.OnTurnAsync(It.IsAny<TurnContext>(), It.IsAny<CancellationToken>())).Callback(() =>
+            {
+            });
+
+            await Assert.ThrowsAsync<Exception>(async () =>
+            {
+                await facebookAdapter.ProcessAsync(httpRequest.Object, httpResponse.Object, bot.Object, default(CancellationToken));
+            });
+        }
+
+        [Fact]
+        public async void ProcessAsyncShouldSucceed()
+        {
+            var payload = File.ReadAllText(Directory.GetCurrentDirectory() + @"\Files\Payload.json");
```

Comment:
Same here on the styling, the variable declarations'd be together with no spacing